### PR TITLE
feat(theme-default): sync tab choices

### PR DESCRIPTION
**Pull request by shulaoda**
## Summary

[Feature]: Syncing PackageManagerTabs and Tabs Choices

## Related Issue

closes: #937 

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

**Comment by Timeless0911** [comment]
We have a `useStorageValue` hook, can you reuse it for your browser judgement and localStorage get/set logic?

https://github.com/web-infra-dev/rspress/blob/89e1c212d47c5a7c3c18987e5c47a9b672d0c119/packages/theme-default/src/logic/useStorageValue.ts#L6-L42

And also the item key `rspress.tabs` can be stored as a CONST and use this const as prefix in code logic, thus reducing maintenance costs~

File: packages/theme-default/src/components/Tabs/index.tsx [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
   useState,
+  useEffect,
+  useContext,
   forwardRef,
   ForwardedRef,
   isValidElement,
   ComponentPropsWithRef,
   type ForwardRefExoticComponent,
 } from 'react';
+import { useStorageValue } from '#theme/logic/useStorageValue';
```
Comment (Timeless0911) [inline packages/theme-default/src/components/Tabs/index.tsx:15]: use relative path is better

Comment (shulaoda) [reply packages/theme-default/src/components/Tabs/index.tsx]: got it

**Comment by Timeless0911** [comment]
LGTM

**Review by Timeless0911 — APPROVED** [review]

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**